app/main/views/agreements.py

```diff
@@ -254,8 +254,9 @@ def approve_for_countersignature(agreement_id):
     framework_agreement_details = framework_agreement.supplier_framework.framework.framework_agreement_details
 
     json_payload = get_json_from_request()
-    json_has_required_keys(json_payload, ['userId'])
-    approved_by_user_id = json_payload['userId']
+    json_has_required_keys(json_payload, ['agreement'])
+    json_has_required_keys(json_payload['agreement'], ['userId'])
```

Comment:
In other routes we only use `json_has_required_keys` to check the top-level object key, and then use `json_has_keys` to check the keys in the nested object.  I'm not 100% sure why it's done like that, but feels like it would be nice to use the same pattern here for consistency, e.g. something like this would match the patter used elsewhere:
```
json_has_required_keys(json_payload, ["agreement"])
update_json = json_payload["agreement"]
json_has_keys(update_json, required_keys=['userId'])
```